### PR TITLE
Bug 1216022 - Clear remote records when a user logout. r=ferjm

### PR DESCRIPTION
**Pull request by weilonge**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=78ec1b5497301390b27fc928d5961cbf1f0be3f4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1536b113bc628965dfe0cfca35b9f8f0679ed728

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3576d5be12a91dd7e920c00a970b41cca6ae160c

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c1645f5896c1d9624de3e66e4fdac727e048b6d4

File: tv_apps/browser/js/sync/db.js [hunk 1]
```diff
@@ -434,6 +434,22 @@ var SyncBrowserDB = {
     // Get a list of bookmarks
     this.db.clearHistoryExcluding(callback);
 
+  },
+
+  clearBookmarks: function browserDB_clearBookmarks(callback) {
```
Comment (ferjm) [inline tv_apps/browser/js/sync/db.js]: `callback` is not used

File: tv_apps/browser/js/sync/db.js [hunk 2]
```diff
@@ -434,6 +434,22 @@ var SyncBrowserDB = {
...
+  },
+
+  clearHistoryDeep: function browserDB_clearHistoryDeep() {
+    return new Promise(resolve => {
+      this.db.clearVisits(resolve);
+    }).then(new Promise(resolve => {
+      this.db.clearPlaces(resolve);
+    })).then(new Promise(resolve => {
+      this.db.clearIcons(resolve);
+    }));
```
Comment (ferjm) [inline tv_apps/browser/js/sync/db.js]: Can we do this in parallel? i.e. Promise.all

File: tv_apps/browser/js/sync/ds_helper.js [hunk 3]
```diff
@@ -58,6 +58,28 @@ SyncDsHelper.prototype = {
...
+      this.registerStoreChangeEvent(() => {
+        this.dataStoreSync(handleTask);
+      });
+      return this.dataStoreSync(handleTask);
+    });
+  },
+
+  stop() {
+    return this.unregisterStoreChangeEvent().then(() => {
+      this.clear();
```
Comment (ferjm) [inline tv_apps/browser/js/sync/ds_helper.js]: `return this.clear()`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b3f8baa7b8588944978fc54693bfe51a7d9b889f

**Pull request merged**